exchanges/binance/binance.go

```diff
@@ -734,41 +743,39 @@ func (b *Binance) SendAuthHTTPRequest(ePath exchange.URL, method, path string, p
 	} else {
 		params.Set("recvWindow", strconv.FormatInt(convert.RecvWindow(recvWindow), 10))
 	}
-	params.Set("recvWindow", strconv.FormatInt(convert.RecvWindow(recvWindow), 10))
-	params.Set("timestamp", strconv.FormatInt(time.Now().Unix()*1000, 10))
-	signature := params.Encode()
-	hmacSigned := crypto.GetHMAC(crypto.HashSHA256, []byte(signature), []byte(b.API.Credentials.Secret))
-	hmacSignedStr := crypto.HexEncodeToString(hmacSigned)
-	headers := make(map[string]string)
-	headers["X-MBX-APIKEY"] = b.API.Credentials.Key
-	if b.Verbose {
-		log.Debugf(log.ExchangeSys, "sent path: %s", path)
-	}
 
-	path = common.EncodeURLValues(path, params)
-	path += "&signature=" + hmacSignedStr
+	ctx, cancel := context.WithTimeout(context.Background(), recvWindow)
+	defer cancel()
+
 	interim := json.RawMessage{}
+	err = b.SendPayload(ctx, f, func() (*request.Item, error) {
+		params.Set("timestamp", strconv.FormatInt(time.Now().Unix()*1000, 10))
+		signature := params.Encode()
+		hmacSigned := crypto.GetHMAC(crypto.HashSHA256, []byte(signature), []byte(b.API.Credentials.Secret))
+		hmacSignedStr := crypto.HexEncodeToString(hmacSigned)
+		headers := make(map[string]string)
+		headers["X-MBX-APIKEY"] = b.API.Credentials.Key
+		path = common.EncodeURLValues(path, params)
```

Comment:
So I've been testing how this works. I'm forcing things to do a second attempt to see what happens. The `fullpath` variable you were using outside the inline function is required upon testing it. Or at least, resetting the `path` var per attempt.
Basically, the context and values remain when retrying, so a `path = common.EncodeURLValues(path, params)` will re-append the values leading to the following type of output where values are duplicated:
```
https://api.binance.com/wapi/v3/depositAddress.html?asset=BTC&recvWindow=10000&status=true&timestamp=1628039926000&signature=569b7f637693510b548d2f5e1a36a5b2e6485de914eee188b5d3f751c920d5e2?asset=BTC&recvWindow=10000&status=true&timestamp=1628039949000&signature=3f9b91345982626a3eaf6d588d8ea6889383051a477599b072b33027bd54c399
```
I'm seeing this style appending on Binance, Bitfinex and Coinbene

I've tested forcing an auth retry on FTX and it works well 🎉

Reply:
🦅 Will fix! Thanks.